solver/llbsolver/ops/build.go

```diff
@@ -117,12 +117,22 @@ func (b *buildOp) Exec(ctx context.Context, inputs []solver.Result) (outputs []s
 	lm.Unmount()
 	lm = nil
 
-	newref, _, err := b.b.Solve(ctx, frontend.SolveRequest{
+	newRes, _, err := b.b.Solve(ctx, frontend.SolveRequest{
 		Definition: def.ToPB(),
 	})
 	if err != nil {
 		return nil, err
 	}
 
-	return []solver.Result{newref}, err
+	var newRef solver.Result
+
+	for k, r := range newRes {
+		if k == "default" {
+			newRef = r
+		} else {
+			r.Release(context.TODO())
```

Comment:
maybe this is obvious, but a comment could be helpful, why release everything except default ?

Reply:
I understand we return the default ref, so we don't want to release it. Anyway, the line just feels obscure to me (but could be because I don't have sufficient knowledge of the codebase yet)